discovery/spec.md

```diff
@@ -365,775 +244,693 @@ The following sections define the attributes that appear in a Service entity.
+
+  This property MAY be serialized as an empty string or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
-  - REQUIRED
-  - MUST be a non-empty array of non-empty strings
+  - OPTIONAL
+  - if present, MUST be a string
+- Examples:
+  - `queue1`
+
+#### authscope
+
+- Type: TBD
+- Description: TBD
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - TBD
 
-##### subscriptionurl
+#### usage
 
-- Type: `URL`
-- Description: An absolute URL indicating where CloudSubscriptions `subscribe`
-  API calls MUST be sent to.
+- Type: String
+- Description: The interaction model supported by this Endpoint. This
+  specification defines a set of possible values (see the
+  [Usage Values](#usage-values) section for more information, however,
+  implementations MAY define new values as extensions. It is RECOMMENDED
+  that extension values be defined with some organizational unique string
+  to reduce the chances of collisions with possible future spec-defined values.
 - Constraints:
   - REQUIRED
+  - MUST be a non-empty string
+- Examples:
+  - `consumer`
+
+#### config
 
-##### subscriptionconfig
+- Type: Object
+- Description: Metadata the provides additional information concerning the
+  interactions with this Endpoint. Each `usage` value MUST define the set of
+  `config` properties that are valid.
 
-- Type: `Map` of `String` to `String`
-- Description: A map indicating supported options for the `config` parameter for
-  the CloudSubscriptions subscribe() API call. Keys are the name of keys in the
-  allowed config map, the values indicate the type of that parameter, conforming
-  to the CloudEvents
-  [type system](../cloudevents/spec.md#type-system).
-  TODO: Needs resolution with CloudSubscriptions API
+  This property MAY be serialized as an empty object or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
   - OPTIONAL
 - Examples:
-  - ??
+  - `{ "protocol": "kafka" , "endpoint": "https://example.com/queue/1234" }`
+
+#### groups
 
-##### subscriptiondialects
+- Type: Array
+- Description: A set of Definition Groups supported by this Endpoint. Each
+  Group MAY be a full representation of the Group or MAY be a
+  [reference](#references) to a Group defined elsewhere. If it is a reference
+  then it MAY be to a Group defined outside of the current Discovery Service
+  instance.
 
-- Type: List of `String` values
-- Description: An array of filter dialects that MAY be used in the
-  Cloud Subscriptions subscribe() API call.
+  This property MAY be serialized as an empty array or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
   - OPTIONAL
+  - if present, the same Group, as identified by its `id` or `uri`, MUST NOT
+    appear more than once within this property.
 - Examples:
-  - `basic`
+  - `[ { ...group properties... }, { "uri": "https://example.com/discovery/group/987", "name", "John's blob store messages" } ]`
```

Comment:
In this example, if the name defined for `https://example.com/discovery/group/987` differs from the name provided by the response from fetching `https://example.com/discovery/group/987`, which one wins? Is this a method for allowing me to overload properties defined in the group for a context of the endpoint?

Reply:
I think that would be an error but in general I'm not sure how we can mandate that they always be the same since both could be correct at the point in time of each query. Meaning, at 12:01 name was "foo" but at 12:02 when someone did a GET to the group it was "bar" because someone modified it during that minute. This is basically true for just about any query, no?